### PR TITLE
Don't require xcodegen settings preset files. All configuration shoul…

### PR DESCRIPTION
**Pull request by rahul-malik**
…d be

specified by xchammer.[yaml|yml] files

### CODE REVIEW

File: Sources/XCHammer/XCBuildSettings.swift [hunk 1]
```diff
@@ -129,9 +129,12 @@ struct Setting<T: XCSettingStringEncodeable & Semigroup>: Semigroup {
     }
 }
 
+
+
+
 struct XCBuildSettings: Encodable {
     var copts: [String] = []
-    var productName: First<String>?
+    var productName: First<String>? = First("$(TARGET_NAME)")
```
Comment (jerrymarino) [inline Sources/XCHammer/XCBuildSettings.swift]: So if we allocate this, it sets it to target name as the default? If we want to always set it to `TARGET_NAME` we should just remove the variable if it isn't necessary

File: Sources/XCHammer/XcodeTarget.swift [hunk 2]
```diff
@@ -454,10 +454,13 @@ public class XcodeTarget: Hashable, Equatable {
             return xcTargetName + ".xctest"
             case .AppExtension, .XPCService, .Watch1App, .Watch2App, .Watch1Extension, .Watch2Extension, .TVAppExtension:
             return xcTargetName + ".appex"
-            case .StaticLibrary:
+            case .StaticLibrary, .Tool:
```
Comment (jerrymarino) [inline Sources/XCHammer/XcodeTarget.swift]: Is this related?

File: Sources/XCHammer/XcodeTarget.swift [hunk 3]
```diff
@@ -454,10 +454,13 @@ public class XcodeTarget: Hashable, Equatable {
...
             return xcTargetName
+            case .Bundle:
+            return xcTargetName + ".bundle"
             default:
-            fatalError()
+            print("ERR: Product type \(productType)")
```
Comment (jerrymarino) [inline Sources/XCHammer/XcodeTarget.swift]: Remove

File: Sources/XCHammer/XcodeTarget.swift [hunk 4]
```diff
@@ -454,10 +454,13 @@ public class XcodeTarget: Hashable, Equatable {
             return xcTargetName + ".xctest"
             case .AppExtension, .XPCService, .Watch1App, .Watch2App, .Watch1Extension, .Watch2Extension, .TVAppExtension:
             return xcTargetName + ".appex"
-            case .StaticLibrary:
+            case .StaticLibrary, .Tool:
             return xcTargetName
+            case .Bundle:
```
Comment (jerrymarino) [inline Sources/XCHammer/XcodeTarget.swift]: Do we actually need this one? I thought bundles weren't built

File: Sources/XCHammer/XcodeTarget.swift [hunk 5]
```diff
@@ -599,8 +602,14 @@ public class XcodeTarget: Hashable, Equatable {
         }
 
         // Product Name
+
         settings.productName <>= self.bundleName.map { First($0) }
 
+        if settings.productName == nil {
+            settings.productName = First(self.extractBuiltProductName())
```
Comment (jerrymarino) [inline Sources/XCHammer/XcodeTarget.swift]: Would this code path ever be hit? It doesn't seem like it if you set the value in the initializer?

File: Sources/XCHammer/XCBuildSettings.swift [hunk 6]
```diff
@@ -163,6 +166,11 @@ struct XCBuildSettings: Encodable {
     var useHeaderMap: First<String>? = First("NO")
     var testTargetName: First<String>?
     var pythonPath: First<String>?
+    var sdkRoot: First<String>?
+    var targetedDeviceFamily: OrderedArray<String> = OrderedArray.empty
+
```
Comment (jerrymarino) [inline Sources/XCHammer/XCBuildSettings.swift]: Minor: remove spaces?

File: Sources/XCHammer/XcodeTarget.swift (hunk 2 above)
Comment (rahul-malik) [reply Sources/XCHammer/XcodeTarget.swift]: Nope but it's correct

File: Sources/XCHammer/XcodeTarget.swift (hunk 4 above)
Comment (rahul-malik) [reply Sources/XCHammer/XcodeTarget.swift]: I actually hit this case

File: Sources/XCHammer/XcodeTarget.swift (hunk 3 above)
Comment (rahul-malik) [reply Sources/XCHammer/XcodeTarget.swift]: done!

File: Sources/XCHammer/XCBuildSettings.swift [hunk 7]
```diff
@@ -163,6 +166,11 @@ struct XCBuildSettings: Encodable {
     var useHeaderMap: First<String>? = First("NO")
     var testTargetName: First<String>?
     var pythonPath: First<String>?
+    var sdkRoot: First<String>?
+    var targetedDeviceFamily: OrderedArray<String> = OrderedArray.empty
```
Comment (kastiglione) [inline Sources/XCHammer/XCBuildSettings.swift]: What's an `OrderedArray`?

**Comment by kastiglione** [comment]
Didn't mean to accept, just meant to leave a comment.

**Comment by kastiglione** [comment]
What's the deal with all the scheme changes and developer machine specific path changes?

File: Sources/XCHammer/XCBuildSettings.swift (hunk 7 above)
Comment (rahul-malik) [reply Sources/XCHammer/XCBuildSettings.swift]: Ordered array is poorly named, it's basically an implementation of ordered set that conform to a few protocols used to serialize build settings. We wanted uniqueness with consistent ordering which is why this was originally written, happy to revisit

File: Sources/XCHammer/XcodeTarget.swift (hunk 5 above)
Comment (rahul-malik) [reply Sources/XCHammer/XcodeTarget.swift]: it will now that there isn't a default value

**Comment by rahul-malik** [comment]
@kastiglione - Filed tickets to address the scheme changes and dev machine paths

**Comment by jerrymarino** [comment]
@rahul-malik - there is a ton of errors on the CI and locally from XcodeGen

```
XCHammer: ** Completed read_aspects in 20.9972s
XCHammer: ** Completed query_genfiles in 2.2947s
XCHammer: Generating project UrlGet
XCHammer: ** Completed compute_deps_hash in 0.3823s
XCHammer: Converting to XcodeGen specification
Unknown product type: genrule. Unable to extract xcode equivalent
Unknown product type: swift_runtime_linkopts. Unable to extract xcode equivalent
Unknown product type: apple_binary. Unable to extract xcode equivalent
Unknown product type: entitlements. Unable to extract xcode equivalent
Unknown product type: apple_binary. Unable to extract xcode equivalent
Unknown product type: entitlements. Unable to extract xcode equivalent
Unknown product type: objc_bundle. Unable to extract xcode equivalent
Unknown product type: swift_runtime_linkopts. Unable to extract xcode equivalent
Unknown product type: swift_runtime_linkopts. Unable to extract xcode equivalent
Unknown product type: apple_binary. Unable to extract xcode equivalent
Unknown product type: _ios_test_bundle. Unable to extract xcode equivalent
Unknown product type: apple_binary. Unable to extract xcode equivalent
Unknown product type: filegroup. Unable to extract xcode equivalent
Unknown product type: swift_runtime_linkopts. Unable to extract xcode equivalent
Unknown product type: entitlements. Unable to extract xcode equivalent
Unknown product type: _ios_test_bundle. Unable to extract xcode equivalent
Unknown product type: swift_runtime_linkopts. Unable to extract xcode equivalent
Unknown product type: apple_binary. Unable to extract xcode equivalent
Unknown product type: swift_runtime_linkopts. Unable to extract xcode equivalent
Unknown product type: apple_binary. Unable to extract xcode equivalent
```

**Comment by rahul-malik** [comment]
I added that print statement, it’s not an error

**Comment by jerrymarino** [comment]
Ah ok - can you submit a followup to remove? People might be confused because we support all of the things that are emitted as `unknown product type`. The function `extractProductType()` is only for Bazel notions that map to Xcode Targets, which many don't.
